pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go

```diff
@@ -0,0 +1,234 @@
+				Spec: v1alpha1.PipelineResourceSpec{
+					Type: v1alpha1.PipelineResourceTypeStorage,
+					Params: []v1alpha1.ResourceParam{{
+						Name: "type", Value: "gcs",
+					}, {
+						Name: "location", Value: "",
+					}},
+				},
+			},
+			want: apis.ErrMissingField("spec.params.location"),
+		}, {
+			name: "invalid resource type",
+			res: &v1alpha1.PipelineResource{
+				Spec: v1alpha1.PipelineResourceSpec{
+					Type: "not-supported",
+				},
+			},
+			want: apis.ErrInvalidValue("spec.type", "not-supported"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.res.Validate(context.Background())
+			if d := cmp.Diff(tt.want.Error(), err.Error()); d != "" {
+				t.Errorf("Didn't get expected error for %s (-want, +got) = %v", tt.name, d)
+			}
+		})
+	}
+}
+
+func TestClusterResourceValidation_Valid(t *testing.T) {
+	tests := []struct {
+		name string
+		res  *v1alpha1.PipelineResource
+	}{
+		{
+			name: "success validate",
+			res: &v1alpha1.PipelineResource{
+				Spec: v1alpha1.PipelineResourceSpec{
+					Type: v1alpha1.PipelineResourceTypeCluster,
+					Params: []v1alpha1.ResourceParam{{
+						Name: "name", Value: "test_cluster_resource",
+					}, {
+						Name: "url", Value: "http://10.10.10.10",
+					}, {
+						Name: "cadata", Value: "bXktY2x1c3Rlci1jZXJ0Cg",
+					}, {
+						Name: "username", Value: "admin",
+					}, {
+						Name: "token", Value: "my-token",
+					}},
+				},
+			},
+		},
+		{
+			name: "specify insecure without cadata",
+			res: &v1alpha1.PipelineResource{
+				Spec: v1alpha1.PipelineResourceSpec{
+					Type: v1alpha1.PipelineResourceTypeCluster,
+					Params: []v1alpha1.ResourceParam{{
+						Name: "name", Value: "test_cluster_resource",
+					}, {
+						Name: "url", Value: "http://10.10.10.10",
+					}, {
+						Name: "username", Value: "admin",
+					}, {
+						Name: "token", Value: "my-token",
+					}, {
+						Name: "insecure", Value: "true",
+					}},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.res.Validate(context.Background()); err != nil {
+				t.Errorf("Unexpected PipelineRun.Validate() error = %v", err)
+			}
+		})
+	}
+}
+
+func TestAllowedGCSStorageType(t *testing.T) {
+	tests := []struct {
+		name        string
+		storageType string
+		want        bool
+	}{
+		{name: "storage with gcs type",
+			storageType: "gcs",
+			want:        true,
+		},
+		{name: "storage with build-gcs type",
+			storageType: "build-gcs",
+			want:        true,
+		},
+		{name: "storage with incorrent type",
+			storageType: "t",
+			want:        false,
+		},
+		{name: "storage with empty type",
+			storageType: "",
+			want:        false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if v1alpha1.AllowedStorageType(tt.storageType) != tt.want {
+				t.Errorf("PipleineResource.allowedStorageType should return %t, got %t for %s", tt.want, !tt.want, tt.storageType)
+			}
+		})
+	}
+
+}
```

Comment:
It seems like some of these files are duplicated vs. moved? e.g. would https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1alpha1/pipelineresource_validation_test.go still exist after this PR?